packages/metal-state/src/validators.js

```diff
@@ -135,7 +144,7 @@ const validators = {
 				}
 				if ((required && !isDefAndNotNull(value[key])) ||
 					isInvalid(validator(value[key]))) {
-					return composeError(ERROR_SHAPE_OF, name, context);
+					return validator(value[key], name + '.' + key, context);
```

Comment:
Hey @diegonvs, Here we could use the literal of ES6.
```javascript
`${name}.${$key}`
```

Reply:
done!